pkg/services/provisioning/dashboards/dashboard.go

```diff
@@ -26,9 +26,10 @@ type DashboardProvisionerFactory func(string) (DashboardProvisioner, error)
 
 // Provisioner is responsible for syncing dashboard from disk to Grafana's database.
 type Provisioner struct {
-	log         log.Logger
-	fileReaders []*FileReader
-	configs     []*config
+	log                log.Logger
+	fileReaders        []*FileReader
+	configs            []*config
+	duplicateValidator *duplicateValidator
```

Comment:
Is there any need for this field to be a pointer? From my testing it can be a normal value type.
```suggestion
	duplicateValidator duplicateValidator
```